### PR TITLE
sdn: move sandbox kill-on-failed-update to runtime socket from direct docker

### PR DESCRIPTION
**Pull request by dcbw**
Follow-on to https://github.com/openshift/origin/pull/14892 that reworks #14665 to use the new runtime socket stuff.

@openshift/networking @danwinship @pravisankar

### CODE REVIEW

**Comment by dcbw** [comment]
cc @sjenning

File: pkg/sdn/plugin/node.go [hunk 1]
```diff
@@ -246,6 +246,28 @@ func ensureDockerClient() (dockertools.DockerInterface, error) {
 	return dockerClient, nil
 }
 
+// This function finds the ContainerID of a failed pod, parses it, and kills
+// any matching Infra container for that pod.
```
Comment (pravisankar) [inline pkg/sdn/plugin/node.go]: Stale comment for this method, needs update as we don't do any parsing, etc.

Comment (dcbw) [inline pkg/sdn/plugin/node.go]: Fixed.

**Comment by kargakis** [comment]
[test]

**Comment by imcsk8** [comment]
LGTM

**Comment by dcbw** [comment]
@pravisankar could I get an lgtm on this one now that we need them?  thanks!

**Comment by kargakis** [comment]
```
--- FAIL: TestOadmPodNetwork (7.05s)
	sdn_test.go:99: could not create namespace "one": could not get NetNamepsace "one": timed out waiting for the condition
	etcd.go:141: dumping etcd to "/openshifttmp/openshift/test-integration/etcd-dump-TestOadmPodNetwork-v2.json"
FAIL
```

**Pull request merged**